src/Connectors/src/ConnectorBase/ServiceInfoCreator.cs

```diff
@@ -15,22 +16,28 @@ namespace Steeltoe.Connector
 {
     public class ServiceInfoCreator
     {
+        private static readonly object _lock = new object();
         private static IConfiguration _config;
         private static ServiceInfoCreator _me = null;
-        private static object _lock = new object();
 
-        internal ServiceInfoCreator(IConfiguration config)
-        {
 #pragma warning disable S3010 // Static fields should not be updated in constructors
```

Comment:
The warning seems valid.  In parallel or concurrent environment, `_config` becomes undetermined.

Reply:
ServiceInfoCreator is supposed to be a static singleton - there should only be one, retrieved via `Instance` below

Reply:
Maybe cleaner to split this class into 2 classes then. One of the factory and one for the configuration.

Reply:
Is latest version better?

Reply:
yep